pg_search/src/postgres/customscan/pdbscan/projections/snippet.rs

```diff
@@ -179,104 +177,150 @@ impl SnippetType {
 struct Context<'a> {
     planning_rti: pg_sys::Index,
     attname_lookup: &'a HashMap<(Varno, pg_sys::AttrNumber), FieldName>,
-    snippet_funcoid: pg_sys::Oid,
-    snippets_funcoid: pg_sys::Oid,
-    snippet_positions_funcoid: pg_sys::Oid,
+    snippet_funcoids: Vec<pg_sys::Oid>,
+    snippets_funcoids: Vec<pg_sys::Oid>,
+    snippet_positions_funcoids: Vec<pg_sys::Oid>,
```

Comment:
So, it's probably a microoptimization, but: all of these `Vec<pg_sys::Oid>` could probably be `[pg_sys::Oid; 2]`: i.e. an array of exactly two elements, which would avoid allocating in a bunch of places.

The array will also be `Copy`, so you won't need to `clone` it.